### PR TITLE
storage: add list gcs objects with s3 library

### PR DESCRIPTION
**Pull request by frankyn**

### CODE REVIEW

File: storage/s3_sdk/list_gcs_objects.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+func listGCSObjects(w io.Writer, bucketName string, googleAccessKeyID string, googleAccessKeySecret string) ([]*s3.Object, error) {
```
Comment (tbpg) [inline storage/s3_sdk/list_gcs_objects.go]: Mind adding sample values for the arguments? Sorry, might also need this for the other sample.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#function-arguments

Comment (tbpg) [inline storage/s3_sdk/list_gcs_objects.go]: Consider dropping the `[]*s3.Object` return value, since it's not used or tested.

Comment (frankyn) [reply storage/s3_sdk/list_gcs_objects.go]: done

Comment (frankyn) [reply storage/s3_sdk/list_gcs_objects.go]: done

File: storage/s3_sdk/list_gcs_buckets.go [hunk 2]
```diff
@@ -27,7 +27,10 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 )
 
-func listGCSBuckets(w io.Writer, googleAccessKeyID string, googleAccessKeySecret string) ([]*s3.Bucket, error) {
+func listGCSBuckets(w io.Writer, googleAccessKeyID string, googleAccessKeySecret string) error {
+	// googleAccessKeyID := fmt.Sprintf("Your Google Access Key ID")
```
Comment (tbpg) [inline storage/s3_sdk/list_gcs_buckets.go]: I don't think we need to use `fmt.Sprintf` for a single string?

File: storage/s3_sdk/list_gcs_buckets.go [hunk 3]
```diff
@@ -42,15 +45,15 @@ func listGCSBuckets(w io.Writer, googleAccessKeyID string, googleAccessKeySecret
 
 	result, err := client.ListBucketsWithContext(ctx, &s3.ListBucketsInput{})
 	if err != nil {
-		return nil, err
+		return err
```
Comment (tbpg) [inline storage/s3_sdk/list_gcs_buckets.go]: Mind adding context to the error?

```
return fmt.Errorf("ListBucketsWithContext: %v", err)
```

That way, you know where the errors came from.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#return-errors

File: storage/s3_sdk/s3_gcs_test.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ func TestBucketList(t *testing.T) {
 	buf := new(bytes.Buffer)
 	err := listGCSBuckets(buf, googleAccessKeyID, googleAccessKeySecret)
 	if err != nil {
-		t.Errorf("listGCSBuckets: %v", err)
+		t.Errorf("%v", err)
```
Comment (tbpg) [inline storage/s3_sdk/s3_gcs_test.go]: Please keep these the same.

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

**Comment by tbpg** [comment]
I updated the objects list test to follow the pattern of the buckets list test and create the test key on the fly. This also has the benefit of not needing the Keystore secrets. Will wait for Kokoro then merge. :)

**Pull request merged**